components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -243,6 +243,13 @@ class P3Microphysics : public AtmosphereProcess
       } // for ipack
       precip_liq_surf_mass(icol) += precip_liq_surf_flux(icol) * PC::RHO_H2O * m_dt;
       precip_ice_surf_mass(icol) += precip_ice_surf_flux(icol) * PC::RHO_H2O * m_dt;
+
+      // Set appropriate boundary fluxes for energy and mass conservation checks.
+      // Any boundary fluxes not included in P3 interface are set to 0.
+      vapor_flux(icol) = 0.0;
+      water_flux(icol) = precip_liq_surf_flux(icol)+precip_ice_surf_flux(icol);
+      ice_flux(icol)   = precip_ice_surf_flux(icol);
```

Comment:
What's the point of `ice_flux`, if it's set to `precip_ice_surf_flux`? Seems like a pointless field?

Reply:
`precip_ice_surf_flux` is in the P3 interface buffer, not in the FM. Also, other processors need to set it to 0 (the checker relies on each process to correctly set all 4 fluxes). I could add `ice_flux` always in P3, and conditionally in SHOC and Radiation.